rust/chains/hyperlane-ethereum/src/contracts/mailbox.rs

```diff
@@ -328,6 +330,70 @@ where
         };
         fill_tx_gas_params(tx, self.provider.clone(), &tx_overrides).await
     }
+
+    async fn simulate_batch(
+        &self,
+        multicall: &mut Multicall<M>,
+        contract_calls: Vec<ContractCall<M, ()>>,
+    ) -> ChainResult<BatchSimulation<M>> {
+        let batch = multicall::batch::<_, ()>(multicall, contract_calls.clone()).await;
+        let batch = self.add_gas_overrides(batch, None).await?;
+        let call_results = batch.call().await?;
+
+        // successful calls within the batch are guaranteed to still be successful after
```

Comment:
at the risk of being pedantic I think this is an assumption of ours and edge cases could exist 😛